### PR TITLE
Added-Leetcode-Arrays-Adhoc-Problem

### PR DESCRIPTION
**Pull request by nishantagrawal01**

### CODE REVIEW

File: src/data/index.js [hunk 1]
```diff
@@ -2139,6 +2139,15 @@ const questions = [
     premium: false,
     companies: ['Amazon', 'Microsoft', 'Bloomberg', 'Apple', 'Adobe'],
   },
+  {
+    id: 2022,
```
Comment (seanprashad) [inline src/data/index.js:2143]: Please change this to `170` - the `id` field isn't the id of the leetcode question in this case

**Review by seanprashad — CHANGES_REQUESTED** [review]
Hey @nishantagrawal01! Thanks for submitting this contribution, however I have a few questions/comments:

1. Can you explain, with an example, what the `Arrays-Adhoc` pattern consists of? I haven't encountered something like this so far in ~300 questions and would love to learn more!
2. According to Leetcode Premium, this question isn't asked by Amazon - did you encounter this recently?

File: src/data/index.js (hunk 1 above)
Comment (nishantagrawal01) [reply src/data/index.js]: Hi @seanprashad , thanks for the info. I have updated the pull req. Kindly please merge it.

**Comment by nishantagrawal01** [comment]
> Hey @nishantagrawal01! Thanks for submitting this contribution, however I have a few questions/comments:
> 
> 1. Can you explain, with an example, what the `Arrays-Adhoc` pattern consists of? I haven't encountered something like this so far in ~300 questions and would love to learn more!
> 2. According to Leetcode Premium, this question isn't asked by Amazon - did you encounter this recently?

Yes Sure. Actually ad-hoc problems are such kind of problems that doesn't basically require particular use of any fancy data structure/algorithm. These problems are also called constructive algorithms and are created in order to build logic even if a candidate is not much into intermediate/advanced level of Data structures/algorithms.

Yes, this was recently asked by amazon in an interview round.

Let me give you an example. Below is the problem statement

Given an array arr[], find the maximum j – i such that arr[j] > arr[i]. 
This is a ad-hoc problem, I couldn't find it on leetcode so I didn't add this one in the list.

**Pull request merged**